ch.sbb.scion.rcp.workbench/src/ch/sbb/scion/rcp/workbench/ISciWorkbenchPopupDialogProvider.java

```diff
@@ -0,0 +1,24 @@
+package ch.sbb.scion.rcp.workbench;
+
+import org.eclipse.jface.dialogs.Dialog;
+import org.eclipse.swt.widgets.Shell;
+
+/**
+ * Allows registering Dialogs in the Eclipse context via the extension point "ch.sbb.scion.rcp.workbench.popupDialogProvider". Classes that
+ * implement this interface must possess a zero-argument constructor, therefore.
+ */
+public interface ISciWorkbenchPopupDialogProvider<T extends Dialog> {
```

Comment:
Do we really want to limit the possible Popups to be based on dialog? In Eclipse itself some are based on Window, especially those which are lightweight or those fancy notication poups and the like.

Reply:
You're right - I am not even using Dialog API but only Window API in the popup interceptor.

Reply:
What holds us back from using IShellProvider? We may invent a shadow interface to provide the missing methods/behaviour, if it is less than what Window would provide.

Reply:
I like the idea. Implemented!